ui/app/src/components/DashboardList/DashboardList.tsx

```diff
@@ -0,0 +1,173 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { DashboardResource } from '@perses-dev/core';
+import { Box, Stack } from '@mui/material';
+import { GridColDef, GridActionsCellItem, GridRowParams, GridValueGetterParams } from '@mui/x-data-grid';
+import DeleteIcon from 'mdi-material-ui/DeleteOutline';
+import PencilIcon from 'mdi-material-ui/Pencil';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { dashboardDisplayName } from '@perses-dev/core/dist/utils/text';
+import { DeleteDashboardDialog } from '../DeleteDashboardDialog/DeleteDashboardDialog';
+import { RenameDashboardDialog } from '../RenameDashboardDialog/RenameDashboardDialog';
+import { DashboardDataGrid, Row } from './DashboardDataGrid';
+
+export interface DashboardListProperties {
+  dashboardList: DashboardResource[];
+  hideToolbar?: boolean;
+}
+
+export function DashboardList(props: DashboardListProperties) {
+  const { dashboardList, hideToolbar } = props;
+
+  const getDashboard = useCallback(
+    (project: string, name: string) => {
+      return dashboardList.find(
+        (dashboard) => dashboard.metadata.project === project && dashboard.metadata.name === name
+      );
+    },
+    [dashboardList]
+  );
+
+  const [rows, setRows] = useState<Row[]>([]);
+  useEffect(() => {
```

Comment:
On my first pass, there’s a lot going on in this component, the `as Row` bit stands out and the number of hooks that I need to wrap my head around, but I need to pull this locally and try it out so I can give some more concrete feedback

Reply:
on my side I didn't see any weird behaviour so far. But perhaps in term of React code, there is something to do to simplify it. But I definitely prefer to rely on expert here :) (@perses/ui ;)) (wondering if this tag is working lol)

Reply:
Oh it looks good now @Nexucis, the way the PR was rebased, we lose out on the previous iterations

@Gladorme in the future, it can be helpful to push additional commits to the branch and leave the old ones so we can look back at the improvements (we squash all the commits when we go to merge the PR)

Reply:
Oh, I just saw your comment after redoing some --amend commits. I will try to do new commit now